src/watcherEx.ts

```diff
@@ -0,0 +1,238 @@
+// Copyright 2022 The Casbin Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { RedisClusterConnection, RedisConnection, RedisClient } from './redis';
+import { ClusterNode, ClusterOptions, RedisOptions } from 'ioredis';
+import { Model, WatcherEx } from 'casbin';
+
+export interface WatcherExOptions extends RedisOptions {
+  channel?: string;
+}
+
+export interface WatcherExClusterOptions extends ClusterOptions {
+  channel?: string;
+}
+
+const updateTypes = {
+  addPolicy: 'updateForAddPolicy',
```

Comment:
`updateFor` seems a little bit redundant. Are there plans to include other messages that would conflict these otherwise?

Reply:
It's just that these are used multiple time. Should I remove it?

Reply:
Actually, let's ensure that this uses the same naming scheme as the golang version does.

Reply:
@Sefriol they are using `UpdateForAddPolicies` like that